src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -107,6 +108,27 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                 }
             }
 
+            public async Task SendRequestContentWithExpect100ContinueAsync(CancellationToken cancellationToken)
+            {
+                // Start timer and try to read response headers.
+                TaskCompletionSource<bool> allowExpect100ToContinue = new TaskCompletionSource<bool>();
+                var expect100Timer = new Timer(
+                            s => ((TaskCompletionSource<bool>)s).TrySetResult(true),
+                            allowExpect100ToContinue, _connection._pool.Settings._expect100ContinueTimeout, Timeout.InfiniteTimeSpan);
+                Task response = ReadResponseHeadersAsync(true, allowExpect100ToContinue);
+
+                // By now, either we got response from server or timer expired.
+                bool sendRequestContent = await allowExpect100ToContinue.Task.ConfigureAwait(false);
+                expect100Timer.Dispose();
```

Comment:
It might be better to wrap this in a 'using' for the timer instead of having an explicit call to .Dispose(). Doing so would allow it to run in an implied 'finally' block.  So, if the code inside the block throws an exception, the timer will definitely be explicitly Dispose'd.

Reply:
updated.